core-tests/shared/src/test/scala/zio/ZLayerSpec.scala

```diff
@@ -582,6 +582,52 @@ object ZLayerSpec extends ZIOBaseSpec {
           layer3    = layer1 ++ layer2
           _        <- layer3.build
         } yield assertCompletes
+      },
+      test("runWith works") {
+        trait TodoRepo {
+          def run(ref: Ref[List[String]]) = ref.update("Running TodoRepo" :: _)
+        }
+
+        object TodoRepo {
+          val layer: ZLayer[Any, Nothing, TodoRepo] = ZLayer.succeed(new TodoRepo {})
```

Comment:
Can have one of the layers / dependencies have a non-Any R requirement? So that we can test that the macro can chain dependencies properly

Reply:
yup - done :)